### ISSUE TITLE
tintin failing to build on darwin

### ISSUE THREAD

**Issue by abathur** [NixOS/nixpkgs#90273]
**Describe the bug**
Tintin started failing to build on macOS at the end of January. I see the same failures local as [appear in hydra](https://hydra.nixos.org/build/121877653). The errors look like:

```
Undefined symbols for architecture x86_64:
  "_fmemopen", referenced from:
      _class_load in class.o
  "_open_memstream", referenced from:
      _class_save in class.o
ld: symbol(s) not found for architecture x86_64
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Makefile:76: tt++] Error 1
builder for '/nix/store/h4lib9n0zhsd99yaba0i0djixxhn76vj-tintin-2.02.01.drv' failed with exit code 2
error: build of '/nix/store/h4lib9n0zhsd99yaba0i0djixxhn76vj-tintin-2.02.01.drv' failed
```

I've tried to do as much legwork as I can, but I don't understand the details well enough to have any sense of what the resolution may be.

1. Here's the [last successful](https://github.com/NixOS/nixpkgs/pull/74684/checks?check_run_id=326615425) check on a version-bump PR, and the [first failed](https://github.com/NixOS/nixpkgs/pull/78884/checks?check_run_id=417891626) check on a version bump PR.
2. It looks like the build started breaking after an update to tintin introduced calls to `open_memstream` and `fmemopen`, which I gather historically didn't exist in libc on macOS (but it sounds like they've been present since 10.13). Here's a link to the diff adding each to tintin:

    - https://github.com/scandum/tintin/commit/4048d5a5ea8b424ed02cd10bdf9ae5f016ba2134#diff-e36164a4409c5c551aac2adcbcbae376R244
    - https://github.com/scandum/tintin/commit/4048d5a5ea8b424ed02cd10bdf9ae5f016ba2134#diff-e36164a4409c5c551aac2adcbcbae376R300
3. I can't swear it's up-to-date, but here are the macOS installation instructions from the official site in case they're meaningful: https://tintin.mudhalla.net/install.php#OSX

**Notify maintainers**
Pinging a few people I've seen dealing with compilation on Darwin. @matthewbauer @LnL7 @vcunat @veprbl @FRidh

**Comment by abathur** [comment]
Many thanks to @alyssais for spotting the obvious, here; it looks like this is about the apple sdk still being at 10.12, while these symbols were introduced in 10.13. 

There was an attempt to update in https://github.com/NixOS/nixpkgs/pull/73744 spurred by the 10.12 URL changing but called off once a new source was found for 10.12

**Comment by veprbl** [comment]
From what I can tell the _open_memstream is available in libsystem_c only starting macOS 10.14.

Perhaps tintin can be patched to work with some independent implementations
* https://www.piumarta.com/software/memstream/
* https://github.com/tmm1/fmemopen
